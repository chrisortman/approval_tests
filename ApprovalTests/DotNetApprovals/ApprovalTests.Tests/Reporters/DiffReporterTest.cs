using ApprovalTests.Reporters;
using NUnit.Framework;

namespace ApprovalTests.Tests.Reporters
{
	[TestFixture]
	[UseReporter(typeof (DiffReporter))]
	public class DiffReporterTest
	{
		private static void AssertLauncher(string approved, string received, DiffReporter reporter)
		{
			var args = reporter.GetLaunchArguments(approved, received);

			try
			{
				Approvals.Approve(args.ToString());
			}
			catch
			{
				reporter.Launch(args);
				throw;
			}
		}

		[Test]
		public void TestLaunchesTortoiseImage()
		{
			AssertLauncher("../../a.png", "../../b.png", new DiffReporter());
		}

		[Test]
		public void TestLaunchesBeyondCompareImage()
		{
			AssertLauncher("../../a.png", "../../b.png", new BeyondCompareReporter());
		}

		[Test]
		public void TestLaunchesTortoiseMerge()
		{
			AssertLauncher("../../a.txt", "../../b.txt", new DiffReporter());
		}

		[Test]
		public void TestWinMerge()
		{
			AssertLauncher("../../a.txt", "../../b.txt", new WinMergeReporter());
		}
	}
}