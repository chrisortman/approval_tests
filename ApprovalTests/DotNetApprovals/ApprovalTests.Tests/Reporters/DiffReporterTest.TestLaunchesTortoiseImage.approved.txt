tortoiseidiff /left:"../../b.png" /right:"../../a.png"